File: src/main/java/redis/embedded/model/ReplicationGroup.java
package redis.embedded.model;

import redis.embedded.core.PortProvider;

import java.util.LinkedList;
import java.util.List;

public final class ReplicationGroup {
    public final String masterName;
    public final int masterPort;
    public final List<Integer> slavePorts = new LinkedList<>();

    public ReplicationGroup(final String masterName, int slaveCount, final PortProvider provider) {
        this.masterName = masterName;
        this.masterPort = provider.get();
        while (slaveCount-- > 0) {
            this.slavePorts.add(provider.get());
        }
    }
}

File: src/main/kotlin/io/github/tobi/laa/embedded/valkey/cluster/highavailability/ReplicationGroup.kt
package io.github.tobi.laa.embedded.valkey.cluster.highavailability

import io.github.tobi.laa.embedded.valkey.ports.PortProvider
import java.util.stream.IntStream.range
import kotlin.streams.toList

data class ReplicationGroup(
    val mainNodeName: String,
    val mainNodePort: Int,
    val replicaPorts: List<Int>
) {

    constructor(mainNodeName: String, provider: PortProvider, replicaCount: Int) : this(
        mainNodeName,
        provider.next(),
        range(0, replicaCount).map { provider.next() }.toList()
    )
}
